packages/ui/src/components/va-counter/VaCounter.vue

```diff
@@ -0,0 +1,391 @@
+    }
+
+    const isMinReached = computed(() => {
+      if (!props.min) { return false }
+      return Number(valueComputed.value) <= props.min
+    })
+
+    const isMaxReached = computed(() => {
+      if (!props.max) { return false }
+
+      return props.step
+        ? Number(valueComputed.value) > (props.max - props.step)
+        : Number(valueComputed.value) >= props.max
+    })
+
+    const disabledDecreaseAction = computed(() => (
+      isMinReached.value || props.readonly || props.disabled
+    ))
+
+    const disabledIncreaseAction = computed(() => (
+      isMaxReached.value || props.readonly || props.disabled
+    ))
+
+    const decreaseCount = () => {
+      if (disabledDecreaseAction.value) { return }
+      calculateCounterValue(Number(valueComputed.value) - props.step)
+    }
+
+    const increaseCount = () => {
+      if (disabledIncreaseAction.value) { return }
+      calculateCounterValue(Number(valueComputed.value) + props.step)
+    }
+
+    const { colorComputed } = useColor(props)
+
+    const decreaseIconProps = computed(() => ({
+      class: { 'va-counter__icon--inactive': disabledDecreaseAction.value },
+      color: colorComputed.value,
+      name: props.decreaseIcon,
+      ...(!disabledDecreaseAction.value && { onClick: decreaseCount }),
+    }))
+
+    const increaseIconProps = computed(() => ({
+      class: { 'va-counter__icon--inactive': disabledIncreaseAction.value },
+      color: colorComputed.value,
+      name: props.increaseIcon,
+      ...(!disabledIncreaseAction.value && { onClick: increaseCount }),
+    }))
+
+    const squareCorners = computed(() => (
+      (typeof props.margins === 'string' ? parseFloat(props.margins) : props.margins) === 0
+    ))
+
+    const decreaseButtonProps = computed(() => ({
+      ...pick(props, ['outline', 'rounded', 'color', 'textColor']),
+      flat: props.flat && !props.outline,
+      outline: props.flat && props.outline,
+      disabled: disabledDecreaseAction.value,
+    }))
+
+    const increaseButtonProps = computed(() => ({
+      ...pick(props, ['rounded', 'color', 'textColor']),
+      flat: props.flat && !props.outline,
+      outline: props.flat && props.outline,
+      disabled: disabledIncreaseAction.value,
+    }))
+
+    const inputAttributesComputed = computed(() => ({
+      ariaLabel: props.label,
+      ...omit(attrs, ['class', 'style']),
+      ...pick(props, ['disabled', 'min', 'max', 'step']),
+      readonly: props.readonly || !props.manualInput,
+    }) as InputHTMLAttributes)
+
+    const inputWrapperPropsComputed = computed(() => ({
+      ...pick(props, ['color', 'readonly', 'disabled', 'messages', 'label', 'bordered', 'outline']),
+    }))
+
+    const classComputed = computed(() => ([
+      attrs.class,
+      { 'va-counter--input-square': squareCorners.value },
+    ]))
+
+    const styleComputed: ComputedRef<Partial<CSSStyleDeclaration>> = computed(() => {
+      const style = attrs.style || {}
+      return Object.assign(style, {
```

Comment:
```suggestion
      return { width: safeCSSLength, ...style }
```

This will create new object, style object will not be modifed.
Style.width will override width from props.

Reply:
refactored